api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -364,6 +369,8 @@ public void changePassword(String pw, String pw2) throws APIException {
 	 * @see org.openmrs.api.UserService#changeHashedPassword(User, String, String)
 	 */
 	public void changeHashedPassword(User user, String hashedPassword, String salt) throws APIException {
+		OpenmrsUtil.validatePassword(user.getUsername(), hashedPassword, user.getSystemId());
```

Comment:
Does this really work? That validate method i think expects non hashed passwords, probably you shouldn't validate this, the ticket says changePassword methods. I also think these should delegate to the changePassword method that validates the password to centralize the code